File: app/src/main/java/com/elementary/tasks/intro/GoogleFixActivity.java
package com.elementary.tasks.intro;

import android.content.Intent;
import android.databinding.DataBindingUtil;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;

import com.elementary.tasks.R;
import com.elementary.tasks.core.ThemedActivity;
import com.elementary.tasks.core.utils.SuperUtil;
import com.elementary.tasks.databinding.ActivityGoogleFixBinding;
import com.elementary.tasks.login.LoginActivity;

public class GoogleFixActivity extends ThemedActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        ActivityGoogleFixBinding binding = DataBindingUtil.setContentView(this, R.layout.activity_google_fix);
        binding.installButton.setOnClickListener(v -> installServices());
    }

    private void installServices() {
        SuperUtil.checkGooglePlayServicesAvailability(this);
    }

    @Override
    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
        super.onActivityResult(requestCode, resultCode, data);
        if (SuperUtil.isGooglePlayServicesAvailable(this)) {
            openLoginScreen();
        }
    }

    private void openLoginScreen() {
        startActivity(new Intent(this, LoginActivity.class));
        finish();
    }
}

File: app/src/main/java/com/elementary/tasks/intro/GoogleFixActivity.kt
package com.elementary.tasks.intro

import android.content.Intent
import android.os.Bundle
import com.elementary.tasks.R
import com.elementary.tasks.core.ThemedActivity
import com.elementary.tasks.core.utils.SuperUtil
import com.elementary.tasks.login.LoginActivity
import com.mcxiaoke.koi.ext.onClick
import kotlinx.android.synthetic.main.activity_google_fix.*

class GoogleFixActivity : ThemedActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_google_fix)
        install_button.onClick { installServices() }
    }

    private fun installServices() {
        SuperUtil.checkGooglePlayServicesAvailability(this)
    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        super.onActivityResult(requestCode, resultCode, data)
        if (SuperUtil.isGooglePlayServicesAvailable(this)) {
            openLoginScreen()
        }
    }

    private fun openLoginScreen() {
        startActivity(Intent(this, LoginActivity::class.java))
        finish()
    }
}
